{
  Exception ex1=createException();
  ex1.initCause(createNestedCause());
  Throwable ex=HiveSQLException.toCause(HiveSQLException.toString(ex1));
  Assert.assertSame(RuntimeException.class,ex.getClass());
  Assert.assertEquals("exception1",ex.getMessage());
  Assert.assertSame(UnsupportedOperationException.class,ex.getCause().getClass());
  Assert.assertEquals("exception2",ex.getCause().getMessage());
  Assert.assertSame(Exception.class,ex.getCause().getCause().getClass());
  Assert.assertEquals("exception3",ex.getCause().getCause().getMessage());
}
